src/dnaio/_core.pyx

```diff
@@ -399,21 +400,30 @@ def record_names_match(header1: str, header2: str):
     # find the spaces and tabs easily.
     cdef char * header1chars = <char *>PyUnicode_1BYTE_DATA(header1)
     cdef char * header2chars = <char *>PyUnicode_1BYTE_DATA(header2)
-    return record_name_bytes_match(header1chars, header2chars)
+    cdef size_t header2length = <size_t>PyUnicode_GET_LENGTH(header2)
+    return record_name_bytes_match(header1chars, header2chars, header2length)
 
 
-cdef bint record_name_bytes_match(char *header1chars, char *header2chars):
+cdef bint record_name_bytes_match(char *header1chars, char *header2chars,
+                                  size_t header2length):
     """
-    Check whether the ascii-encoded names match.
+    Check whether the ascii-encoded names match. Only header2length is needed.
     """
     # Only the first part (i.e. the name without the comment) is of interest.
```

Comment:
When I added the `--rename` option to Cutadapt, I had to decide on a term for "the part of the read header before the first whitespace" and settled on "id" (or read id). I’d like to use these terms consistently in dnaio as well at some point, so you could write "Only the read ID is of interest".

Reply:
I renamed everything to "id". It makes the function easier to read too.